Explorer/Assets/DCL/Passport/Modules/EquippedItems_PassportModuleController.cs

```diff
@@ -0,0 +1,316 @@
+
+            LayoutRebuilder.ForceRebuildLayoutImmediate(view.EquippedItemsContainer);
+            LayoutRebuilder.ForceRebuildLayoutImmediate(scrollContainer);
+        }
+
+        private int CalculateMissingEmptyItems(int totalItems)
+        {
+            int remainder = totalItems % 6;
+            int missingItems = (remainder == 0) ? 0 : 6 - remainder;
+            return missingItems;
+        }
+
+        private async UniTaskVoid AwaitEquippedItemsPromiseAsync(WearablePromise equippedWearablesPromise, EmotePromise equippedEmotesPromise, CancellationToken ct)
+        {
+            var wearablesUniTaskAsync = await equippedWearablesPromise.ToUniTaskAsync(world, cancellationToken: ct);
+            var emotesUniTaskAsync = await equippedEmotesPromise.ToUniTaskAsync(world, cancellationToken: ct);
+
+            if (!wearablesUniTaskAsync.Result!.Value.Succeeded || !emotesUniTaskAsync.Result!.Value.Succeeded || ct.IsCancellationRequested)
+                return;
+
+            var currentWearables = wearablesUniTaskAsync.Result.Value.Asset.Wearables;
+            var currentEmotes = emotesUniTaskAsync.Result.Value.Asset.Emotes;
+            SetGridElements(currentWearables, currentEmotes);
+        }
+
+        private async UniTaskVoid WaitForThumbnailAsync(IWearable itemWearable, EquippedItem_PassportFieldView itemView, CancellationToken ct)
+        {
+            Sprite? sprite = await thumbnailProvider.GetAsync(itemWearable, ct);
+
+            if (ct.IsCancellationRequested)
+                return;
+
+            itemView.EquippedItemThumbnail.sprite = sprite;
+        }
+
+        private async UniTaskVoid WaitForThumbnailAsync(IEmote itemEmote, EquippedItem_PassportFieldView itemView, CancellationToken ct)
+        {
+            Sprite? sprite = await thumbnailProvider.GetAsync(itemEmote, ct);
+
+            if (ct.IsCancellationRequested)
+                return;
+
+            itemView.EquippedItemThumbnail.sprite = sprite;
+        }
+
+        private void ClearLoadingItems()
+        {
+            foreach (EquippedItem_PassportFieldView loadingItem in instantiatedLoadingItems)
+                loadingItemsPool.Release(loadingItem);
+
+            instantiatedLoadingItems.Clear();
+        }
+
+        private void ClearEquippedItems()
+        {
+            foreach (EquippedItem_PassportFieldView equippedItem in instantiatedEquippedItems)
+                equippedItemsPool.Release(equippedItem);
+
+            instantiatedEquippedItems.Clear();
+        }
+
+        private void ClearEmptyItems()
+        {
+            foreach (EquippedItem_PassportFieldView emptyItem in instantiatedEmptyItems)
+                emptyItemsPool.Release(emptyItem);
+
+            instantiatedEmptyItems.Clear();
+        }
+
+        private static string GetMarketplaceLink(string id)
+        {
+            const string MARKETPLACE = "https://market.decentraland.org/contracts/{0}/items/{1}";
+            string[] split = id.Split(":");
```

Comment:
Please consider using `Span` and its non-allocating API

Reply:
Done: https://github.com/decentraland/unity-explorer/pull/1277/commits/e0f043ab58fe4c48f90bd5956c4f2cf285d251d4